### PR TITLE
docs(msq): update insert vs replace for dimension-based segment pruning

### PR DESCRIPTION
**Pull request by 317brian**
### Description

Updates the docs to be more explicit

Release note: n/a

<hr>

<hr>

This PR has:

- [x] been self-reviewed.

### CODE REVIEW

File: docs/multi-stage-query/concepts.md [hunk 1]
```diff
@@ -102,6 +104,19 @@ issues](./known-issues.md#select) page.
 
 For more information about the syntax, see [REPLACE](./reference.md#replace).
 
+When deciding whether to use REPLACE or INSERT, keep in mind that REPLACE can perform dimension-based segment pruning but INSERT cannot. To use dimension-based segment pruning, your query needs to meet the following requirements:
+
+- Segments were generated by a REPLACE statement, not an INSERT statement.
```
Comment (kfaraz) [inline docs/multi-stage-query/concepts.md:109]: I don't think this section needs to be reiterated here as this info is already captured later in the "Clustering" section. I think just linking to that section is enough.

File: docs/multi-stage-query/concepts.md [hunk 2]
```diff
@@ -102,6 +104,19 @@ issues](./known-issues.md#select) page.
 
 For more information about the syntax, see [REPLACE](./reference.md#replace).
 
+When deciding whether to use REPLACE or INSERT, keep in mind that REPLACE can perform dimension-based segment pruning but INSERT cannot. To use dimension-based segment pruning, your query needs to meet the following requirements:
```
Comment (kfaraz) [inline docs/multi-stage-query/concepts.md:107]: This part and the above change on line 81/82 needs to be rephrased.

To clarify, INSERT or REPLACE queries (better called statements for less ambiguity in this case?) cannot and need not perform dimension-based pruning (* unless they are doing a sub-SELECT on some other datasource).
It is the queries (i.e. SELECT queries of any kind) that we run on the _segments_ generated with the above INSERT or REPLACE statements that could potentially benefit from such pruning.

As a suggestion:

```suggestion
When deciding whether to use REPLACE or INSERT, keep in mind that segments generated with REPLACE can be pruned with dimension-based pruning but those generated with INSERT cannot. Please refer to [Clustering](#clustering) for the prerequisites of dimension-based pruning.:
```

**Review by kfaraz — CHANGES_REQUESTED** [review]
Requested some changes.

**Comment by 317brian** [comment]
Thanks for the review @kfaraz ! I'll take a look at your comments this week. (I didn't get a notification email for them for some reason, so apologies for the late response.)

File: docs/multi-stage-query/concepts.md (hunk 2 above)
Comment (317brian) [reply docs/multi-stage-query/concepts.md:107]: Linking to it is probably a better solution than straight repeating. Will make this change

**Review by kfaraz — APPROVED** [review]
Thanks for the improvement, @317brian !

**Pull request merged**